### ISSUE TITLE
🐛Cloudflared Update From 2024.8.3 to 2024.9.0 in K8s Keeps Crashing the Pod

### ISSUE THREAD

**Issue by GGSHAH** [cloudflare/cloudflared#1326]
**Description:**

Hi Everyone,

I've encountered an issue  when version updated from 2024.8.3 to 2024.9.0 automatically, which causes the Kubernetes pod running the cloudflared service to crash. Upon further investigation, I could not find any documentation or release notes related to this behavior in the version updates.

To prevent the pod from crashing, I had to include the `--no-autoupdate` flag after going through the docs for troubleshooting. However, this workaround is not ideal in the long term, and I believe more users could face this issue. Moreover, I could not find any information in the docs when deploying for [Kubernetes environment.](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/deploy-tunnels/deployment-guides/kubernetes/)

**Steps to Reproduce:**

 Deploy a Cloudflare service on a Kubernetes pod using version 2024.8.3 & It gets automatically updated and the pod crashes after it is updated to 2024.9.0.

Workaround: Adding the `--no-autoupdate` flag prevents the pod from crashing after the update.

Expected Behavior: The pod should not crash after updating to the latest version.

Actual Behavior: The pod crashes after the update to 2024.9.0 without any information in the documentation about this behavior.

**Environment:**

    Kubernetes version: AWS EKS 1.30
    Cloudflare version: 2024.9.0
    Deployment method: Helm

Request: Please provide clarification in the documentation or release notes regarding this issue and a long-term fix, if possible.

**Logs and errors**
```
2024-09-10T13:21:48Z INF Version 2024.9.0
2024-09-10T13:21:48Z INF GOOS: linux, GOVersion: go1.22.2, GoArch: amd64
2024-09-10T13:21:48Z INF Settings: map[metrics:0.0.0.0:2000 token:*****]
2024-09-10T13:21:48Z INF Autoupdate frequency is set autoupdateFreq=86400000
2024-09-10T13:21:48Z INF Generated Connector ID: ***
2024-09-10T13:21:48Z INF Initial protocol quic
2024-09-10T13:21:48Z INF ICMP proxy will use *** as source for IPv4
2024-09-10T13:21:48Z INF ICMP proxy will use *** in zone eth0 as source for IPv6
2024-09-10T13:21:48Z WRN The user running cloudflared process has a GID (group ID) that is not within ping_group_range. You might need to add that user to a group within that range, or instead update the range to encompass a group the user is already in by modifying /proc/sys/net/ipv4/ping_group_range. Otherwise cloudflared will not be able to ping this network error="Group ID 65532 is not between ping group 1 to 0"
2024-09-10T13:21:48Z WRN ICMP proxy feature is disabled error="cannot create ICMPv4 proxy: Group ID 65532 is not between ping group 1 to 0 nor ICMPv6 proxy: socket: permission denied"
2024-09-10T13:21:48Z INF Starting metrics server on [::]:2000/metrics
2024/09/10 13:21:48 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.
2024-09-10T13:21:49Z ERR Failed to create new quic connection error="failed to dial to edge with quic: context canceled" connIndex=2 event=0 ip=***
2024-09-10T13:21:49Z INF Registered tunnel connection connIndex=0 connection=f04f68e6-6329-4bb1-b939-5d66ea76e797 event=0 ip=198.41.192.107 location=lhr09 protocol=quic
2024-09-10T13:21:49Z INF Registered tunnel connection connIndex=1 connection=28610a10-a755-4674-94f2-2b08a1b8c1a4 event=0 ip=198.41.200.73 location=lhr13 protocol=quic
2024-09-10T13:21:50Z ERR Failed to create new quic connection error="failed to dial to edge with quic: context canceled" connIndex=3 event=0 ip=198.41.192.7
2024-09-10T13:21:50Z INF Registered tunnel connection connIndex=2 connection=9b224ed6-a173-49f7-9e3a-29187726ad4e event=0 ip=198.41.200.33 location=lhr13 protocol=quic
2024-09-10T13:21:51Z INF Tunnel server stopped
cloudflared has been updated to version 2024.9.0
```

**Comment by jcsf** [comment]
Hi @GGSHAH , container images are meant to be immutable, so the `--no-autoupdate` is a requirement when running in environments like docker or kubernetes. We will update the [Kubernetes environment](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/deploy-tunnels/deployment-guides/kubernetes/) accordingly.

Regarding of why you are hitting this issue right now, can be one of many reasons:
- You are using our official image with `latest` tag but your Kubernetes cluster isn't respecting the `imagePullPolicy: Always` that it should assume if it finds a `latest` tag. This would force every time the pod restart for the image to be pulled and therefore updated to the latest release, avoiding the auto-update to kick in. Otherwise, it will keep using the old image which is in an old version.
- You are not using our official image, and therefore the providers of your image didn't update their latest image to our latest release yet, so the auto-update kicks in.

Regardless, **my recommendation would be that you add the `--no-autoupdate` to the command on the manifest file, avoid using the latest tag and use a specific version instead**. Then you should have some **automatic system like renovate or dependabot updating the image on your manifest to get the latest versions**. Another alternative is adding the `imagePullPolicy: Always` let the pod kill itself when there is an update, because it will then pull the new image and start working again and you get automatic updates, **note that THIS ISN'T AT ALL a controlled way of doing rolling upgrades**.

You can see more information regarding images on Kubernetes here:
https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting

**Comment by jcsf** [comment]
We just found out that there was an issue with the binaries of 2024.9.0 that was causing it to continuously auto-update. Your issue was caused by that, so actually my option 1 _"You are using our official image with latest tag but your Kubernetes cluster isn't respecting t..."_ was wrong, it was the binary that was faulty, not Kubernetes. We are releasing the 2024.9.1 which will fix that and after that 2024.9.2 with the new improvements fixed.

Regardless, what I said regarding container images still applies, so we will still update the documentation.

**Comment by GGSHAH** [comment]
> so actually my option 1 "You are using our official image with latest tag but your Kubernetes cluster isn't respecting t..." was wrong
 
 Yeah I checked it as well. My Image Policy is set to `imagePullPolicy: Always` already. 
>  We just found out that there was an issue with the binaries of 2024.9.0 that was causing it to continuously auto-update. Your issue was caused by that

That makes sense why we did not face any issue before this version.
> Regardless, what I said regarding container images still applies, so we will still update the documentation.

Thanks !
